dtls.c

```diff
@@ -853,7 +870,7 @@ static int verify_ext_eliptic_curves(uint8 *data, size_t data_length) {
   data += sizeof(uint16);
   if (i + sizeof(uint16) != data_length) {
     dtls_warn("the list of the supported elliptic curves should be tls extension length - 2\n");
-    return dtls_alert_fatal_create(DTLS_ALERT_HANDSHAKE_FAILURE);
+    return 1;
```

Comment:
My mistake - I have reverted back these changes.  For some reason I thought that `dtls_alert_fatal_create()` was actually generating an alert rather than just returning a negative error.